lib/insights_cloud/async/insights_generate_notifications.rb

```diff
@@ -0,0 +1,58 @@
+require 'rest-client'
+
+module InsightsCloud
+  module Async
+    class InsightsGenerateNotifications < ::Actions::EntryAction
+      # cache blueprint on class level, so it won't be reloaded on subsequent calls
+      def self.blueprint
+        @blueprint ||= NotificationBlueprint.find_by(name: 'insights_satellite_hits')
+      end
+
+      def run
+        add_satellite_notifications
+      end
+
+      def add_satellite_notifications
+        hits_count = InsightsHit.where(host_id: foreman_host.id).count
+
+        # Remove stale notifications
+        blueprint.notifications.destroy_all
+
+        if hits_count > 0
+          add_notification(hits_count)
+        end
+      end
+
+      private
+
+      def logger
+        action_logger
+      end
+
+      def foreman_host
+        ForemanRhCloud.foreman_host
+      end
+
+      def blueprint
+        self.class.blueprint
+      end
+
+      def add_notification(hits_count)
+        Notification.create!(
+          initiator: User.anonymous_admin,
+          audience: ::Notification::AUDIENCE_ADMIN,
+          message: UINotifications::StringParser.new(blueprint.message, {subject: hits_count}).to_s,
```

Comment:
should it be:
```suggestion
          message: UINotifications::StringParser.new(blueprint.message, {hits_count: hits_count}).to_s,
```

and then in the [message](https://github.com/theforeman/foreman_rh_cloud/pull/540/files#diff-cd6cf07e82a109107cad0037db33477d19d1e8ad65fb3973cea2ccef81f49274R5) it will be more readable

Reply:
Done. This was kind of leftover from earlier version: in theory `message` can be generated automatically by the [framework](https://github.com/theforeman/foreman/blob/3347fa49d500964f0209122d8d36c920d1feafcc/app/models/notification.rb#L79), but the framework injects only the `subject` and `initiator`. Unfortunately `subject` has to be an ActiveRecord, so I had to parse the string manually.